app/scripts/metamask-controller.js

```diff
@@ -600,6 +606,43 @@ export default class MetamaskController extends EventEmitter {
         this.platform.showTransactionNotification(txMeta, rpcPrefs);
 
         const { txReceipt } = txMeta;
+
+        // if this is a transferFrom method generated from within the app it may be a collectible transfer transaction
+        // in which case we will want to check and update ownership status of the transferred collectible.
+        if (
+          txMeta.type === TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER_FROM &&
+          txMeta.txParams !== undefined
+        ) {
+          const {
+            data,
+            to: contractAddress,
+            from: userAddress,
+          } = txMeta.txParams;
+          const { chainId } = txMeta;
+          const transactionData = getTransactionData(data);
+          const tokenAmountOrTokenId = getTokenValueParam(transactionData);
+          const { allCollectibles } = this.collectiblesController.state;
+
+          // check if its a known collectible
+          const knownCollectible = allCollectibles?.[userAddress]?.[
+            chainId
+          ].find(
+            ({ address, tokenId }) =>
+              isEqualCaseInsensitive(address, contractAddress) &&
+              tokenId === tokenAmountOrTokenId,
+          );
+
+          // if it is we check and update ownership status.
+          if (knownCollectible) {
+            this.collectiblesController.checkAndUpdateSingleCollectibleOwnershipStatus(
+              knownCollectible,
+              false,
+              // TODO add this when checkAndUpdateSingleCollectibleOwnershipStatus is updated
+              // { userAddress, chainId },
```

Comment:
This should/will be added bac this PR lands https://github.com/MetaMask/controllers/pull/672 and a new controllers version is cut.

Reply:
We can add this after this PR is merged, correct?

Reply:
What affect will this call to `this.collectiblesController.checkAndUpdateSingleCollectibleOwnershipStatus` have before this code is uncommented?

Reply:
Yea we can add it later. Until the PR is merged and this code is uncommented the following scenario is possible:
- User sends NFT on Account 1
- User switches to Account 2 before transaction is complete
- Transaction is confirmed while user is on Account 2
- The collectibles controller will check whether this collectible is owned by Account 2 and update its status, but the NFT was owned by Account 1 so it won't find it and no status will be updated.

Reply:
Without the PR merged and code uncommented (in its current state) If however the user hasn't switched accounts between when they sent the NFT and when the transaction was confirmed, the status update will be completed correctly.